assets/src/form-submit.ts

```diff
@@ -0,0 +1,267 @@
+import { type SubmittedFileValidationRequest } from "../../src/validation/submit.controller.validation";
```

Comment:
I think the code looks good, but it may be worth adding some comments in the file to describe what the functions are doing, as it's quite a lot of sometimes confusing code

Reply:
Is there anything sepcifically you find find confusing?

Reply:
I think in particular it's where each function is called in the overall process, to make it easier for a dev to follow the process from when a user uploads till the validation is complete and the redirect happens.